scipy/interpolate/_bsplines.py

```diff
@@ -445,34 +476,87 @@ def integrate(self, a, b, extrapolate=None):
         if extrapolate is None:
             extrapolate = self.extrapolate
 
-        if not extrapolate:
-            # shrink the integration interval, if needed
-            a = max(a, self.t[self.k])
-            b = min(b, self.t[-self.k - 1])
+        # Prepare self.t and self.c.
+        self._ensure_c_contiguous()
 
-            if self.c.ndim == 1:
-                # fast path: use FITPACK's routine (cf _fitpack_impl.splint)
-                t, c, k = self.tck
-                aint, wrk = _dierckx._splint(t, c, k, a, b)
-                return aint
+        # Swap integration bounds if needed.
+        sign = 1
+        if b < a:
+            a, b = b, a
+            sign = -1
 
-        # prepare t & c
-        self._ensure_c_contiguous()
+        out = np.empty((2, prod(self.c.shape[1:])), dtype=self.c.dtype)
+        integral = np.empty((1, prod(self.c.shape[1:])), dtype=self.c.dtype)
```

Comment:
Since `integral.fill(0)` failed in one of the tests, I realized that it has to be declared before the starting of the if-else block. Should I declare it elsewhere?